pkg/api/option.go

```diff
@@ -159,12 +148,27 @@ func WithTimeout(timeout time.Duration) Option {
 	}
 }
 
+// WithHostname sets the X-Machine-Name header to the passed in hostname.
+func WithHostname(hostname string) Option {
+	return func(c *Client) {
+		next := c.doFunc
+		c.doFunc = func(c *Client, req *http.Request) (*http.Response, error) {
+			hostname = url.QueryEscape(strings.TrimSpace(hostname))
```

Comment:
It should be done at `params.go` after this [line](https://github.com/wakatime/wakatime-cli/blob/develop/cmd/params/params.go#L264). The `hostname` is always parsed once when reading from input and like other variables we already "clean up/fomat" them.

Reply:
No, we should encode headers when making the request not when parsing params. This encoding is only for over the wire so we shouldn't do it early.